packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts

```diff
@@ -97,13 +98,20 @@ function defineHasOneRelationship(
   relationship: RelationshipMetadata,
   relatedEntity: DmlEntity<
     Record<string, SchemaType<any> | RelationshipType<any>>
-  >
+  >,
+  hooks: EntityHooks
 ) {
   const relatedModelName = upperCaseFirst(relatedEntity.name)
+  const shouldRemoveRelated = !!hooks.deleted?.remove.includes(
+    relationship.name
+  )
+
   OneToOne({
     entity: relatedModelName,
     nullable: relationship.nullable,
     mappedBy: relationship.mappedBy as any,
+    onDelete: shouldRemoveRelated ? "cascade" : undefined,
+    cascade: shouldRemoveRelated ? (["soft-remove"] as any) : undefined,
```

Comment:
Let's discuss after, but here be careful, the soft remove should be applied on the other side, also, specifying cascade remove the defaults like persists for example